### PR TITLE
smpp (unbind replace cancel) encode and decode.

### PR DESCRIPTION
**Pull request by Lico-Tom**

### CODE REVIEW

File: netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppConst.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 public class SmppConst {
 
+
```
Comment (Shoothzj) [inline netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppConst.java:24]: unnecessary blank line

File: netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java [hunk 2]
```diff
@@ -264,6 +277,36 @@ private ByteBuf encodeDeliverSmResp(ChannelHandlerContext ctx, SmppDeliverSmResp
...
+        writeCString(buf, message.body().serviceType());
+        writeCString(buf, message.body().messageId());
+        buf.writeByte(message.body().sourceAddrTon());
+        buf.writeByte(message.body().sourceAddrTon());
+        writeCString(buf, message.body().sourceAddr());
+        buf.writeByte(message.body().destAddrNpi());
+        buf.writeByte(message.body().destAddrTon());
+        writeCString(buf, message.body().destinationAddr());
+        return buf;
+    }
```
Comment (Shoothzj) [inline netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java:309]: add a blan line after this

File: netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java [hunk 3]
```diff
@@ -294,6 +337,14 @@ private ByteBuf encodeBindTransceiverResp(ChannelHandlerContext ctx, SmppBindTra
...
     }
 
+
+    private ByteBuf encodeOutBind(ChannelHandlerContext ctx, SmppOutBind message) {
+        ByteBuf buf = ctx.alloc().buffer(SmppConst.LEN_OUT_BIND_MSG);
+        writeHeader(buf, message.header(), SmppConst.LEN_OUT_BIND_MSG);
+        writeCString(buf, message.body().systemId());
+        writeCString(buf, message.body().password());
+        return buf;
+    }
```
Comment (Shoothzj) [inline netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java:347]: add a blan line after this

File: netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java [hunk 4]
```diff
@@ -294,6 +337,14 @@ private ByteBuf encodeBindTransceiverResp(ChannelHandlerContext ctx, SmppBindTra
         return buf;
     }
 
+
```
Comment (Shoothzj) [inline netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java:340]: remove this blank line

Comment (Lico-Tom) [reply netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java:340]: OK，fix completed

File: netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java (hunk 3 above)
Comment (Lico-Tom) [reply netty-codec-smpp/src/main/java/io/github/protocol/codec/smpp/SmppEncoder.java:347]: OK，fix completed

**Review by Shoothzj — APPROVED** [review]

**Pull request merged**